src/modules/workspaces/workspaces.usecase.ts

```diff
@@ -707,6 +724,53 @@ export class WorkspacesUsecases {
     return newMember;
   }
 
+  async getMemberDetails(
+    workspaceId: WorkspaceAttributes['id'],
+    memberId: User['uuid'],
+  ) {
+    const workspace = await this.workspaceRepository.findById(workspaceId);
+
+    if (!workspace) {
+      throw new BadRequestException('Invalid workspace');
+    }
+
+    const [workspaceUser, user] = await Promise.all([
+      this.workspaceRepository.findWorkspaceUser({
+        workspaceId: workspace.id,
+        memberId,
+      }),
+      this.userRepository.findByUuid(memberId),
+    ]);
+
+    if (!workspaceUser || !user) {
+      throw new BadRequestException('Invalid user');
+    }
+
+    const workspaceTeamUser =
+      await this.teamRepository.getTeamAndMemberByWorkspaceAndMemberId(
+        workspace.id,
+        workspaceUser.memberId,
+      );
+
+    return {
+      workspaceUser,
```

Comment:
We may merge this data with the user's object so we make things as clear and easy as possible for any API's consumer